definitions.h

```diff
@@ -232,6 +232,10 @@ class Attributes
             // Your code goes here when clipping is implemented
         }
         PIXEL color;
+        double attr[16]; // to be more dynamic, just use an array of attributes that would be of type
```

Comment:
I decided to use a map here instead of an array. That way you have a viable way to reference which attribute values you want without having to maintain a pattern or fixed hidden rule for organizing the values.